src/braket/circuits/circuit.py

```diff
@@ -440,22 +441,38 @@ def _check_if_qubit_measured(
         Raises:
             ValueError: If adding a gate or noise operation after a measure instruction.
         """
-        if (
-            # check if there is a measure instruction on the target qubit
-            target
-            and target in self._measure_targets
-            # check if there is a measure instruction on any qubits in the target_mapping
-            or (target_mapping and any(targ in self._measure_targets for targ in target_mapping))
-            # If no target or target_mapping is supplied, check if there is a measure
-            # instruction on the current instructions target qubit
-            or (
-                instruction.target
-                and any(targ in self._measure_targets for targ in instruction.target)
+        if self._measure_targets:
+            measure_on_target = (
+                target and any(tar in self._measure_targets for tar in target)
+                if isinstance(target, Iterable)
+                else target in self._measure_targets
```

Comment:
Can this be combined into lines 453-457? An `Instruction` containing a `Measure` operator is still an instruction.

Reply:
Never mind, `instruction` and `target` are decoupled, so this still makes sense.